plugin/core/message_request_handler.py

```diff
@@ -0,0 +1,93 @@
+import mdpopups
+from .rpc import Client
+from .typing import Any, List, Callable
+from .protocol import Response
+from sublime import View
+
+
+class MessageRequestHandler(object):
+    def __init__(self, view: View, client: Client, request_id: Any, params: dict) -> None:
+        self.client = client
+        self.request_id = request_id
+        self.request_sent = False
+        self.view = view
+        actions = params.get("actions", [])
+        self.titles = list(action.get("title") for action in actions)
+        self.message = params.get('message', '')
+        self.message_type = params.get('type', 4)
+
+    def _send_user_choice(self, href: int = -1) -> None:
+        if not self.request_sent:
+            self.request_sent = True
+            self.view.hide_popup()
+            # when noop; nothing was selected e.g. the user pressed escape
+            param = None
+            index = int(href)
+            if index != -1:
+                param = {"title": self.titles[index]}
+            response = Response(self.request_id, param)
+            self.client.send_response(response)
+
+    def show(self) -> None:
+        show_notification(
+            self.view,
+            self.message_type,
+            self.message,
+            self.titles,
+            self._send_user_choice,
+            self._send_user_choice
+        )
+
+
+def message_content(message_type: int, message: str, titles: List[str]) -> str:
+    formatted = []
+    icons = {
+      1: '❗',
+      2: '⚠️',
+      3: 'ℹ️',
+      4: '📝'
+    }
+    icon = icons.get(message_type, '')
+    formatted.append("<p class='message'>{} {}</p>".format(icon, message))
+
+    buttons = []
+    for idx, title in enumerate(titles):
+        buttons.append("<a href='{}'>{}</a>".format(idx, title))
+
+    formatted.append("<p class='actions'>" + " ".join(buttons) + "</p>")
+
+    return "".join(formatted)
+
+
+def show_notification(view: View, message_type: int, message: str, titles: List[str],
+                      on_navigate: Callable, on_hide: Callable) -> None:
+    myStyle = """
```

Comment:
Can we name this variable `css` instead of `myStyle`? Try to avoid camelCase naming and instead go for snake_case naming.